src/Microsoft.ML.Data/Evaluators/Metrics/RankingMetrics.cs

```diff
@@ -40,16 +42,14 @@ internal RankingMetrics(IExceptionContext ectx, DataViewRow overallResult)
         {
             VBuffer<double> Fetch(string name) => Fetch<VBuffer<double>>(ectx, overallResult, name);
 
-            DiscountedCumulativeGains = Fetch(RankingEvaluator.Dcg).GetValues().ToArray();
-            NormalizedDiscountedCumulativeGains = Fetch(RankingEvaluator.Ndcg).GetValues().ToArray();
+            DiscountedCumulativeGains = Fetch(RankingEvaluator.Dcg).DenseValues().ToImmutableArray();
+            NormalizedDiscountedCumulativeGains = Fetch(RankingEvaluator.Ndcg).DenseValues().ToImmutableArray();
         }
 
         internal RankingMetrics(double[] dcg, double[] ndcg)
         {
-            DiscountedCumulativeGains = new double[dcg.Length];
-            dcg.CopyTo(DiscountedCumulativeGains, 0);
-            NormalizedDiscountedCumulativeGains = new double[ndcg.Length];
-            ndcg.CopyTo(NormalizedDiscountedCumulativeGains, 0);
+            DiscountedCumulativeGains = dcg.ToImmutableArray();
```

Comment:
Do we _need_ to do a copy here? Would it be terrible if we just set the field to an array?

Reply:
I'm honestly not certain, but as I indicated in the description I think it's better to be more concerned at this moment with getting the right public surface. If it turns out that this is an area where we can do an optimization to avoid a copy later, I'm all for it, but I doubt given the nature of this code that a copy will really wind up making a big difference here (as opposed to, say, something in the actual data pipeline).